src/combinatorics/catalan.lean

```diff
@@ -129,3 +144,54 @@ by norm_num [catalan_eq_central_binom_div, nat.central_binom, nat.choose]
 
 lemma catalan_three : catalan 3 = 5 :=
 by norm_num [catalan_eq_central_binom_div, nat.central_binom, nat.choose]
+
+namespace tree
+open_locale tree
+
+/-- Given two finsets, find all trees that can be formed with
+  left child in `a` and right child in `b` -/
+@[simp] def pairwise_node (a : finset (tree unit)) (b : finset (tree unit)) : finset (tree unit) :=
```

Comment:
Why `simp`?

Reply:
The other options are:
  - Inline the definition: This makes the definition of `trees_of_num_nodes_eq` harder to read and makes some of the simp calls run slightly more slowly (e.g. on line 170, although the goal still closes even without the `-pairwise_node`).
  - Remove the `simp` attribute: I found that typically the right thing to do is just unfold this, except when it is slow (and you don't need to). Otherwise, I feel like I would have to make an API for it, and I'm not sure it will be used enough (or anywhere else really) to merit an API.

So marking it as `simp` seemed to me like a reasonable compromise.

Reply:
You can also make it `reducible`.

Reply:
That worked surprisingly well! I changed it to reducible.